src/server/analytic_services/prometheus_reporting.js

```diff
@@ -144,6 +159,50 @@ async function start_server(
             }
         }
     }
+
+    while (retry_https_count) {
+        try {
+            if (ssl_port > 0 && config.ALLOW_HTTPS_METRICS) {
+                const ssl_cert_info = await ssl_utils.get_ssl_cert_info('METRICS', nsfs_config_root);
```

Comment:
Can we re-use start_server_and_cert() like the other services?

Reply:
@romayalon We can reuse `start_server_and_cert()` only because of the `endpoint_request_handler` hold all the endpoint for s3, sts and iam. That is not the case for prometheus, prometheus handler is defined in `prometheus_reporting.js` and for reusing the `start_server_and_cert()` we need to move the `metrics_request_handler` to endpoint.js but that again add the condition block inside `start_server_and_cert()` and mix prometheus and endpoint code.

Reply:
moving `metrics_request_handler` to the `endpoint.js` is not the solution we should pursue...
I can give you an idea of how to start the changes that - 
1. rename `start_server_and_cert()` to `start_endpoint_server_and_cert()`
2. extract from `start_server_and_cert()` the HTTP start server, and the HTTPS start server into 2 new functions and locate them in http_utils.
3. rename `promehteus_reporting.start_server()` to `start_metrics_server()`
4. call the new 2 HTTP and HTTPS servers start from `start_metrics_server()`

let me know what you think and if you need help in this refactoring